lib/error-handler.js

```diff
@@ -62,13 +64,31 @@ function handleError (reply, error, cb) {
     return
   }
 
-  const result = func(error, reply.request, reply)
-  if (result !== undefined) {
-    if (result !== null && typeof result.then === 'function') {
-      wrapThenable(result, reply)
-    } else {
-      reply.send(result)
+  try {
+    const result = func(error, reply.request, reply)
+    if (result !== undefined) {
+      if (result !== null && typeof result.then === 'function') {
+        wrapThenable(result, reply)
+      } else {
+        reply.send(result)
+      }
+    }
+  } catch (err) {
+    reply[kReplyIsError] = true
+    try {
+      if (err.cause) {
+        throw FST_ERR_ERROR_CAUSE_ALREADY_PRESENT()
```

Comment:
We throw an Error, to abort the control flow. But we instantiate an Error and generate a stacktrace, which we probably dont need anyway?
We dont need it to be in the try catch in this case.

Reply:
Are you suggesting we could do this instead if err.cause is already set, without entering the try/catch?:

```
reply.log.warn({
  err: FST_ERR_ERROR_CAUSE_ALREADY_PRESENT(),
  parentError: err,
  reason: failedToAssignError,
  message: 'Failed to assign child error as cause to parent error thrown from error handler'
})
```